### ISSUE TITLE
Double quotes on dump and import failing

### ISSUE THREAD

**Issue by back-2-95** [Smile-SA/gdpr-dump#145]
We have two Drupal projects. Another gives working dump.sql but another one gives dump which used double quotes.
Configuration files are same. 

### Preconditions

GdprDump Version: 5.0.2

PHP Version: 8.3

Database Version: MySQL 8.0.30 in DBaaS

### Steps to reproduce

1. ./gdpr-dump.phar conf/gdpr-dump.yaml > dump.sql

### Expected result

```sql
CREATE TABLE `batch` (
  `bid` int unsigned NOT NULL AUTO_INCREMENT COMMENT 'Primary Key: Unique batch ID.',
  `token` varchar(64) CHARACTER SET ascii COLLATE ascii_general_ci NOT NULL COMMENT 'A string token generated against the current user''s session id and the batch id, used to ensure that only the user who submitted the batch can effectively access it.',
  `timestamp` bigint NOT NULL COMMENT 'A Unix timestamp indicating when this batch was submitted for processing. Stale batches are purged at cron time.',
  `batch` longblob COMMENT 'A serialized array containing the processing data for the batch.',
  PRIMARY KEY (`bid`),
  KEY `token` (`token`)
) 
```

### Actual result

```sql
CREATE TABLE "batch" (
  "bid" int unsigned NOT NULL AUTO_INCREMENT COMMENT 'Primary Key: Unique batch ID.',
  "token" varchar(64) CHARACTER SET ascii COLLATE ascii_general_ci NOT NULL COMMENT 'A string token generated against the current user''s session id and the batch id, used to ensure that only the user who submitted the batch can effectively access it.',
  "timestamp" bigint NOT NULL COMMENT 'A Unix timestamp indicating when this batch was submitted for processing. Stale batches are purged at cron time.',
  "batch" longblob COMMENT 'A serialized array containing the processing data for the batch.',
  PRIMARY KEY ("bid"),
  KEY "token" ("token")
);
```

and an error then importing:

```
ERROR 1064 (42000) at line 27: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '"batch" (
    "bid" int unsigned NOT NULL AUTO_INCREMENT COMMENT 'Primary Key: Uni' at line 1
```

**Comment by amenk** [comment]
Can you try to isolate the problem? What is the difference between the projects? Are all versions the same? Did you try to reproduce it in a fresh `mysql:8.0.30` docker container or so?

**Comment by back-2-95** [comment]
The one which works has MySQL 8.4.2 in a Docker container.
The one which does not work (double quotes) has MySQL 8.0.30 in DBaaS

**Comment by back-2-95** [comment]
I need to setup a local setup with docker containers with different versions and see if I can reproduce this that way.

**Comment by guvra** [comment]
Check the value of the SQL mode in your database, maybe it is set to something like `ANSI_QUOTES`?

**Comment by back-2-95** [comment]
Database as a Service has this SQL mode:

```
ANSI,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION,NO_ZERO_DATE,NO_ZERO_IN_DATE,STRICT_ALL_TABLES
```

**Comment by guvra** [comment]
It seems like ANSI_QUOTES is enabled then:

[ANSI](https://dev.mysql.com/doc/refman/8.4/en/sql-mode.html#sqlmode_ansi)
Equivalent to [REAL_AS_FLOAT](https://dev.mysql.com/doc/refman/8.4/en/sql-mode.html#sqlmode_real_as_float), [PIPES_AS_CONCAT](https://dev.mysql.com/doc/refman/8.4/en/sql-mode.html#sqlmode_pipes_as_concat), **[ANSI_QUOTES](https://dev.mysql.com/doc/refman/8.4/en/sql-mode.html#sqlmode_ansi_quotes)**, [IGNORE_SPACE](https://dev.mysql.com/doc/refman/8.4/en/sql-mode.html#sqlmode_ignore_space), and [ONLY_FULL_GROUP_BY](https://dev.mysql.com/doc/refman/8.4/en/sql-mode.html#sqlmode_only_full_group_by).

**Comment by back-2-95** [comment]
Hmm, so this is I guess something which is configures on MySQL side and my options are:

```
ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION,NO_ZERO_DATE,NO_ZERO_IN_DATE,STRICT_ALL_TABLES
```

or

```
REAL_AS_FLOAT,PIPES_AS_CONCAT,IGNORE_SPACE,ONLY_FULL_GROUP_BY,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION,NO_ZERO_DATE,NO_ZERO_IN_DATE,STRICT_ALL_TABLES
```

Wondering what all is affected if I do the change

**Comment by back-2-95** [comment]
Confirmed this with local Docker image and settings same SQL mode there. So with that I got the same dump as from the DBaaS

**Comment by back-2-95** [comment]
I just can't totally get my brain around this:

- Our previous solution is also based on https://github.com/druidfi/mysqldump-php v1
- Same database cluster, with same `ANSI` there in the sql_mode
- `mysqldump-php` is just triggered via Drush command https://github.com/druidfi/drush-gdpr-dumper

🧐

**Comment by back-2-95** [comment]
UPDATE: I have discovered that this is somehow related to some change which has happened to database cluster on 10. - 11. of this month.

**Comment by back-2-95** [comment]
Somehow mysqldump-php ends up changing backtics to double quotes.

Here in the screenshot:

- Left is mysqldump from mariadb used
- right is mysqldump-php used

![image](https://github.com/user-attachments/assets/555c3e38-e2b2-42d6-bb18-0b94ceb09524)

**Comment by back-2-95** [comment]
If I add to mysqldump-php:

```
SET SESSION sql_mode='REAL_AS_FLOAT,PIPES_AS_CONCAT,IGNORE_SPACE,ONLY_FULL_GROUP_BY,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION,NO_ZERO_DATE,NO_ZERO_IN_DATE,STRICT_ALL_TABLES'
```

Dump will be "forced" to be right, without changing sql_mode on server

**Comment by back-2-95** [comment]
@amenk & @guvra would you give me comments for https://github.com/druidfi/mysqldump-php/pull/53/files ?

**Comment by guvra** [comment]
> [@amenk](https://github.com/amenk) & [@guvra](https://github.com/guvra) would you give me comments for https://github.com/druidfi/mysqldump-php/pull/53/files ?

I don't think you need to implement a workaround, there's already an option available to set SQL mode:

```
dump:
    init_commands:
        - 'SET SESSION SQL_MODE="REAL_AS_FLOAT,PIPES_AS_CONCAT,IGNORE_SPACE,ONLY_FULL_GROUP_BY,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION,NO_ZERO_DATE,NO_ZERO_IN_DATE,STRICT_ALL_TABLES'
```

**Comment by back-2-95** [comment]
Yes, I can confirm it works as intended and can work as fix whoever finds this as a problem 👍
